### PR TITLE
Change `source` to `.` when calling startup scripts in shell runner

### PR DESCRIPTION
**Pull request by heyealex**
Update the startup script shell runner invocation from source, which is specific to bash, to `.` which is compatible with more shells. 

A set of new test_configs have been added that test startup scripts against different OS images. This change has been deployed and tested against these new test_configs.

### Submission Checklist

* [x] Have you installed and run this change against pre-commit? (`pre-commit
  install`)
* [x] Are all tests passing? (`make tests`)
* [x] Have you written unit tests to cover this change?
* [x] Is unit test coverage still above 80%?
* [x] Have you updated all applicable documentation?
* [x] Have you followed the guidelines in our Contributing document?

### CODE REVIEW

File: modules/scripts/startup-script/templates/startup-script-custom.tpl [hunk 1]
```diff
@@ -29,7 +29,7 @@ stdlib::runner() {
   case "$1" in
     ansible-local) stdlib::run_playbook "$destpath/$filename" "$args";;
     # shellcheck source=/dev/null
-    shell)  sh -c "source '$destpath/$filename' $args";;
+    shell)  sh -c ". '$destpath/$filename' $args";;
```
Comment (cboneti) [inline modules/scripts/startup-script/templates/startup-script-custom.tpl:32]: Here, on line 13, I would add a set -x so it fails when the runner fails. 
Also, I recommend replacing line 13 by: 
  . '$destpath/$filename' $args"
  exit_code=$?
  stdlib::debug "=== $object finished with exit_code=$exit_code ==="

**Comment by tpdownes** [comment]
I think we also need to add documentation that is very explicit about what type of shell scripts we support. I never interpreted it to mean "POSIX" but that is the status quo.

File: modules/scripts/startup-script/templates/startup-script-custom.tpl [hunk 2]
```diff
@@ -29,7 +29,7 @@ stdlib::runner() {
   case "$1" in
     ansible-local) stdlib::run_playbook "$destpath/$filename" "$args";;
     # shellcheck source=/dev/null
```
Comment (nick-stroud) [inline modules/scripts/startup-script/templates/startup-script-custom.tpl:31]: What is this line doing? Do we still need `source=/dev/null` now that you are using `.`?

**Comment by heyealex** [comment]
A note as I'm running through some tests:
* Using `.` to source within `sh -c` in ubuntu (and probably debian) means arguments aren't passed to the script.
* Using `.` to source without `sh -c` can have some odd effects with commands like `exit 1`. The entire process will exit with a failure in this case. That may be what we want, but it's worth noting as I ran into it during testing.

File: modules/scripts/startup-script/templates/startup-script-custom.tpl (hunk 1 above)
Comment (heyealex) [reply modules/scripts/startup-script/templates/startup-script-custom.tpl]: Done

File: modules/scripts/startup-script/templates/startup-script-custom.tpl (hunk 2 above)
Comment (heyealex) [reply modules/scripts/startup-script/templates/startup-script-custom.tpl]: Removed

**Review by cboneti — APPROVED** [review]

**Comment by heyealex** [comment]
> I think we also need to add documentation that is very explicit about what type of shell scripts we support. I never interpreted it to mean "POSIX" but that is the status quo.

The recent change removed the `sh -c`, so the shell used will be the default shell, and not strictly a POSIX shell.

**Comment by heyealex** [comment]
The integration test failure is one that has been exposed by this change, Omnia is not installing properly anymore ands needs to be updated to a more recent version.

**Pull request merged**